test/e2e.sh

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env bash
+
+set -Eeuxo pipefail
+
+readonly REPO_ROOT="${REPO_ROOT:-$(git rev-parse --show-toplevel)}"
+
+run_ct_container() {
+    echo 'Running ct container...'
+    docker run --rm --interactive --detach --network host --name test-cont \
+        --volume "$(pwd):/go/src/github.com/mattermost/mattermost-operator" \
+        --workdir "/go/src/github.com/mattermost/mattermost-operator" \
+        "golang:1.12.2" \
+        cat
+    echo
+}
+
+docker_exec() {
+    docker exec --interactive test-cont "$@"
+}
+
+
+run_kind() {
+    echo "Download kind binary..."
+    curl -sSLo kind https://github.com/kubernetes-sigs/kind/releases/download/"${KIND_VERSION}"/kind-linux-amd64
+    chmod +x kind
+    sudo mv kind /usr/local/bin/kind
+
+    kind --version
+
+    echo "Download kubectl..."
+    curl -sSLo kubectl https://storage.googleapis.com/kubernetes-release/release/"${K8S_VERSION}"/bin/linux/amd64/kubectl
+    chmod +x kubectl
+    sudo cp kubectl /usr/local/bin/
+    docker cp kubectl test-cont:/usr/local/bin/
```

Comment:
Is this meant to be run locally or just for CI? I wouldn't want it to use sudo or mess with my existing kubectl, kind, kubeconfig, etc.

Reply:
on ci only for local, i will implement another call